clients/s3/tableid.go

```diff
@@ -0,0 +1,40 @@
+package s3
+
+import (
+	"fmt"
+
+	"github.com/artie-labs/transfer/lib/config/constants"
+	"github.com/artie-labs/transfer/lib/sql"
+)
+
+type TableIdentifier struct {
+	database string
+	schema   string
+	table    string
+}
+
+func NewTableIdentifier(database, schema, table string) TableIdentifier {
+	return TableIdentifier{database: database, schema: schema, table: table}
+}
+
+func (ti TableIdentifier) Database() string {
+	return ti.database
+}
+
+func (ti TableIdentifier) Schema() string {
+	return ti.schema
+}
+
+func (ti TableIdentifier) Table() string {
+	return ti.table
+}
+
+func (ti TableIdentifier) FullyQualifiedName(_, uppercaseEscNames bool) string {
```

Comment:
Compare to https://github.com/artie-labs/transfer/pull/450/files#diff-25a5472065a253195427b0d5cca5e99df35797c1291b52cd6687d670a585b500L44